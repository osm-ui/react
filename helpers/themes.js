import _ from 'lodash';
import { lighten } from 'polished';


export function buildDarkThemeConfig(baseConfig, colors, colorVariant) {
    const color1 = colors[`${colorVariant}1`];
    const color2 = colors[`${colorVariant}2`];
    const color3 = colors[`${colorVariant}3`];
    // const color4 = colors[`${colorVariant}4`];
    const color5 = colors[`${colorVariant}5`];

    return _.merge({}, baseConfig, {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color1,
        loaderColor: colors.white,
        controlColor: color1,
        hoverControlColor: colors.white,

        column: {
            nav: {
                color: colors.white,
                backgroundColor: lighten(0.1, color2),
                hoverColor: colors.white,
                hoverBackgroundColor: color3,
            },
        },
        toolbar: {
            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        titlebar: {
            color: colors.white,
            backgroundColor: color2,
            borderColor: color2,

            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        form: {
            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color5,
                hoverBackgroundColor: color3,
                hoverBorderColor: color5,
                focusBackgroundColor: color3,
                focusBorderColor: color5,
                activeBackgroundColor: color3,
                activeBorderColor: color5,

                primary: {
                    color: color2,
                    backgroundColor: colors.white,
                    borderColor: colors.white,
                    hoverBackgroundColor: colors.lightGray5,
                    hoverBorderColor: colors.lightGray5,
                    focusBackgroundColor: colors.lightGray5,
                    focusBorderColor: colors.lightGray5,
                    activeBackgroundColor: colors.lightGray5,
                    activeBorderColor: colors.lightGray5,
                },
            },
        },
    });
}


export function buildLightThemeConfig(baseConfig, colors, colorVariant) {
    const color1 = colors[`${colorVariant}1`];
    const color2 = colors[`${colorVariant}2`];
    const color3 = colors[`${colorVariant}3`];
    const color4 = colors[`${colorVariant}4`];
    const color5 = colors[`${colorVariant}5`];

    return _.merge({}, baseConfig, {
        color: colors.anthracite1,
        backgroundColor: color2,
        borderColor: color1,
        loaderColor: colors.anthracite1,
        controlColor: color4,
        hoverControlColor: colors.anthracite1,

        column: {
            nav: {
                color: colors.anthracite1,
                backgroundColor: lighten(0.2, color2),
                hoverColor: colors.anthracite1,
                hoverBackgroundColor: color3,
            },
        },
        toolbar: {
            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        titlebar: {
            color: colors.anthracite1,
            backgroundColor: color2,
            borderColor: color2,

            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        form: {
            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color5,
                hoverBackgroundColor: color3,
                hoverBorderColor: color5,
                focusBackgroundColor: color3,
                focusBorderColor: color5,
                activeBackgroundColor: color3,
                activeBorderColor: color5,

                primary: {
                    color: color2,
                    backgroundColor: colors.anthracite1,
                    borderColor: colors.anthracite1,
                    hoverBackgroundColor: colors.lightGray5,
                    hoverBorderColor: colors.lightGray5,
                    focusBackgroundColor: colors.lightGray5,
                    focusBorderColor: colors.lightGray5,
                    activeBackgroundColor: colors.lightGray5,
                    activeBorderColor: colors.lightGray5,
                },
            },
        },
    });
}


export default {
    buildDarkThemeConfig,
    buildLightThemeConfig,
};
