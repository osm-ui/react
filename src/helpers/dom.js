export function isOverflowYHidden(element) {
  return window.getComputedStyle(element).overflowY !== 'hidden';
}
