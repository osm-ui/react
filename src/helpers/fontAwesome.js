export function makePrefix(style) {
  return `fa${style[0]}`;
}

export function makeName(name) {
  return `fa-${name}`;
}
