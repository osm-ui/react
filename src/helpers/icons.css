@import url('https://use.fontawesome.com/releases/v5.2.0/css/all.css');
