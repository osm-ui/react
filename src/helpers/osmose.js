export function formatDeletedTags(fix, osmTags) {
  return {
    ...fix,
    del: fix.del.map(deletedTag => {
      return osmTags.find(tag => tag.k === deletedTag);
    })
  };
}

export function formatOsmTags(data) {
  return data.reduce((acc, item) => {
    return {
      ...acc,
      [item.k]: item.v
    };
  }, {});
}

export function fixOsmTags(osmTags, fix) {
  const newOsm = {
    ...formatOsmTags(osmTags),
    ...formatOsmTags(fix.add),
    ...formatOsmTags(fix.mod)
  };

  fix.del.forEach(tag => Reflect.deleteProperty(newOsm, tag));

  return newOsm;
}
