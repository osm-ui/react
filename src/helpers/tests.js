import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import DefaultTheme from 'themes/Default';

export const snapshotWithElementChildren = (TestedElement, props) => {
  const result = shallow(
    <DefaultTheme>
      <TestedElement {...props}>
        <div />
      </TestedElement>
    </DefaultTheme>
  );

  expect(result.childAt(0)).toMatchSnapshot();
};

export const snapshotWithTextChildren = (TestedElement, props) => {
  const result = shallow(
    <DefaultTheme>
      <TestedElement {...props}>This is a text</TestedElement>
    </DefaultTheme>
  );

  expect(result.childAt(0)).toMatchSnapshot();
};
