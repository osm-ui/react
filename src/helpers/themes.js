import _ from 'lodash';
import { lighten, darken } from 'polished';
import { injectGlobal } from 'styled-components';

export function buildDarkThemeConfig(baseConfig, colors, colorVariant) {
  const color1 = colors[`${colorVariant}1`];
  const color2 = colors[`${colorVariant}2`];
  const color3 = colors[`${colorVariant}3`];
  // const color4 = colors[`${colorVariant}4`];
  const color5 = colors[`${colorVariant}5`];

  /* eslint-disable no-unused-expressions */
  injectGlobal`
    .osm-ui-react-marker-${colorVariant} {
      color: ${colors.white};

      #colorized, #colorized * {
        fill: ${color2} !important;
      }
    }
  `;
  /* eslint-enable */

  return _.merge({}, baseConfig, {
    color: colors.white,
    backgroundColor: color2,
    borderColor: color1,
    loaderColor: colors.white,
    controlColor: color1,
    hoverControlColor: colors.white,

    sidebar: {
      nav: {
        color: colors.white,
        backgroundColor: lighten(0.1, color2),
        hoverColor: colors.white,
        hoverBackgroundColor: color3
      }
    },
    toolbar: {
      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    titlebar: {
      color: colors.white,
      backgroundColor: color2,
      borderColor: color2,

      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    form: {
      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color5,
        hoverBackgroundColor: color3,
        hoverBorderColor: color5,
        focusBackgroundColor: color3,
        focusBorderColor: color5,
        activeBackgroundColor: color3,
        activeBorderColor: color5,

        primary: {
          color: color2,
          backgroundColor: colors.white,
          borderColor: colors.white,
          hoverBackgroundColor: colors.lightGray5,
          hoverBorderColor: colors.lightGray5,
          focusBackgroundColor: colors.lightGray5,
          focusBorderColor: colors.lightGray5,
          activeBackgroundColor: colors.lightGray5,
          activeBorderColor: colors.lightGray5
        }
      }
    }
  });
}

export function buildLightThemeConfig(baseConfig, colors, colorVariant) {
  const color1 = colors[`${colorVariant}1`];
  const color2 = colors[`${colorVariant}2`];
  const color3 = colors[`${colorVariant}3`];
  const color4 = colors[`${colorVariant}4`];
  const color5 = colors[`${colorVariant}5`];
  const darkColor1 = darken(0.27, color1);

  /* eslint-disable no-unused-expressions */
  injectGlobal`
        .osm-ui-react-marker-${colorVariant} {
            color: ${darkColor1};

            #colorized, #colorized * {
                fill: ${color2} !important;
            }
        }
    `;
  /* eslint-enable */

  return _.merge({}, baseConfig, {
    color: darkColor1,
    backgroundColor: color2,
    borderColor: color1,
    loaderColor: darkColor1,
    controlColor: color4,
    hoverControlColor: darkColor1,

    sidebar: {
      nav: {
        color: darkColor1,
        backgroundColor: color3,
        hoverColor: darkColor1,
        hoverBackgroundColor: lighten(0.14, color3)
      }
    },
    toolbar: {
      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    titlebar: {
      color: darkColor1,
      backgroundColor: color2,
      borderColor: color2,

      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    form: {
      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color5,
        hoverBackgroundColor: color3,
        hoverBorderColor: color5,
        focusBackgroundColor: color3,
        focusBorderColor: color5,
        activeBackgroundColor: color3,
        activeBorderColor: color5,

        primary: {
          color: color2,
          backgroundColor: darkColor1,
          borderColor: darkColor1,
          hoverBackgroundColor: colors.lightGray5,
          hoverBorderColor: colors.lightGray5,
          focusBackgroundColor: colors.lightGray5,
          focusBorderColor: colors.lightGray5,
          activeBackgroundColor: colors.lightGray5,
          activeBorderColor: colors.lightGray5
        }
      }
    }
  });
}

export default {
  buildDarkThemeConfig,
  buildLightThemeConfig
};
