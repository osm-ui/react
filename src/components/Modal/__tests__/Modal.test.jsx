import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import { snapshotWithElementChildren } from 'helpers/tests';
import { Modal } from 'index';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Modal));
});

describe('When using snapshots with title, header and footer', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Modal, {
      title: 'title',
      header: <div />,
      footer: <div />
    }));
});

describe('When using snapshots with loader', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Modal, {
      loading: true,
      loaderLabel: 'loading'
    }));
});

describe('When testing the callbacks', () => {
  const onOpen = jest.fn();

  it('Should call the onOpen callback when component is mounted', () => {
    shallow(<Modal onOpen={onOpen} />);
    expect(onOpen.mock.calls.length).toBe(1);
  });

  // TODO: test onClose
});
