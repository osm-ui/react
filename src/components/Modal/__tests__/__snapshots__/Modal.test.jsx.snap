// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with an element children 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
}

.c1 {
  z-index: 1000;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 3rem;
  background: rgba(255,255,255,0.95);
  color: #101113;
  overflow-y: auto;
  opacity: 0;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
  position: absolute;
}

.c1.scroll-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: none;
}

.c1.scroll-content .header {
  margin-bottom: 2rem;
}

.c1 .content {
  padding: 2rem;
}

.c1.scroll-content .content {
  overflow-y: auto;
  margin: 0;
  border-color: #dbdcdd;
  border-style: solid;
  border-width: 1px 0 1px 0;
}

.c1.scroll-content .content::after {
  content: '';
  margin-top: 2rem;
  display: block;
}

.c1 .content.loading {
  display: none;
}

.c1.scroll-content .footer {
  margin-top: 2rem;
}

<span
  className="c0"
>
  <aside
    className="container-root c1"
    style={undefined}
  >
    
    <div
      className="content"
    >
      <div />
    </div>
    
  </aside>
</span>
`;

exports[`When using snapshots with loader Should render with an element children 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
}

.c3 {
  width: 4rem;
  height: 4rem;
  margin: 0 auto;
  border-width: 3px;
  border-style: solid;
  border-color: #707377;
  border-radius: 50%;
  border-left-color: transparent;
  border-bottom-color: transparent;
  border-right-color: transparent;
  -webkit-animation: iVXCSc 650ms infinite linear;
  animation: iVXCSc 650ms infinite linear;
}

.c2 {
  text-align: center;
}

.c2.centered {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  color: #707377;
  margin-top: 1.5rem;
}

.c1 {
  z-index: 1000;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 3rem;
  background: rgba(255,255,255,0.95);
  color: #101113;
  overflow-y: auto;
  opacity: 0;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
  position: absolute;
}

.c1.scroll-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: none;
}

.c1.scroll-content .header {
  margin-bottom: 2rem;
}

.c1 .content {
  padding: 2rem;
}

.c1.scroll-content .content {
  overflow-y: auto;
  margin: 0;
  border-color: #dbdcdd;
  border-style: solid;
  border-width: 1px 0 1px 0;
}

.c1.scroll-content .content::after {
  content: '';
  margin-top: 2rem;
  display: block;
}

.c1 .content.loading {
  display: none;
}

.c1.scroll-content .footer {
  margin-top: 2rem;
}

<span
  className="c0"
>
  <aside
    className="container-root c1"
    style={undefined}
  >
    
    <div
      className="content loading"
    >
      <div />
    </div>
    <div
      className="centered c2"
    >
      <div>
        <div
          className="c3"
        />
        <div
          className="c4"
        >
          loading
        </div>
      </div>
    </div>
  </aside>
</span>
`;

exports[`When using snapshots with title, header and footer Should render with an element children 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
}

.c2 {
  margin: -3rem 0 5rem;
}

.c2.in-header {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin: 0 0 0 2rem;
  line-height: 5rem;
}

.c1 {
  z-index: 1000;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 3rem;
  background: rgba(255,255,255,0.95);
  color: #101113;
  overflow-y: auto;
  opacity: 0;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
  position: absolute;
}

.c1.scroll-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: none;
}

.c1.scroll-content .header {
  margin-bottom: 2rem;
}

.c1 .content {
  padding: 2rem;
}

.c1.scroll-content .content {
  overflow-y: auto;
  margin: 0;
  border-color: #dbdcdd;
  border-style: solid;
  border-width: 1px 0 1px 0;
}

.c1.scroll-content .content::after {
  content: '';
  margin-top: 2rem;
  display: block;
}

.c1 .content.loading {
  display: none;
}

.c1.scroll-content .footer {
  margin-top: 2rem;
}

<span
  className="c0"
>
  <aside
    className="container-root c1"
    style={undefined}
  >
    <header
      className="header"
    >
      <h2
        className="sidebar-title in-header c2"
      >
        title
      </h2>
      <div />
    </header>
    <div
      className="content"
    >
      <div />
    </div>
    <div />
  </aside>
</span>
`;
